x-pack/legacy/plugins/lens/public/indexpattern_plugin/operations/definitions/date_histogram.tsx

```diff
@@ -166,28 +144,43 @@ export const dateHistogramOperation: OperationDefinition<DateHistogramIndexPatte
     const intervalIsRestricted =
       field!.aggregationRestrictions && field!.aggregationRestrictions.date_histogram;
 
-    function intervalToNumeric(interval: string) {
-      return supportedIntervals.indexOf(interval);
-    }
+    const interval = parseInterval(currentColumn.params.interval);
 
-    function numericToInterval(i: number) {
-      return supportedIntervals[i];
-    }
+    // We force the interval value to 1 if it's empty, since that is the ES behavior,
+    // and the isValidInterval function doesn't handle the empty case properly. Fixing
+    // isValidInterval involves breaking changes in other areas.
+    const isValid = isValidInterval(
+      `${interval.value === '' ? '1' : interval.value}${interval.unit}`,
+      restrictedInterval(field!.aggregationRestrictions) as string
```

Comment:
We always cast the actual return type `string | undefined` here to a `string` when we use this. As someone not that familiar with the code, it's not clear to me why we could do that? Why do we know that that method will always return a string, even though it could also return` undefined`? Is there potentially a better way to type that? If not, can we at least leave a comment, why we know this can never return `undefined`, or build an assert into that `restrictedInterval` function that will throw if it ever would be (and TS would then properly detect that it's only a `string`)?

Reply:
It's because the type signature for restrictedInterval is wrong. It requires a string, but handles undefined explicitly and properly. I just didn't want to have a post-feature-freeze PR affect code outside of Lens, if possible, so I didn't fix the type signature.